### ISSUE TITLE
Error installing mod_wsgi

### ISSUE THREAD

**Issue by Darrius-Me** [GrahamDumpleton/mod_wsgi#579]
I am trying to install mod_wsgi using:
    **pip install mod_wsgi**
and prompt an error
  **error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/**

I already set my Apache home folder to the apache folder installed via Wamp Server 64. I already have the x86 and x64 of **Microsoft Visual C++ 2015-2019 Redistributable** version 14.22.27281.0, Visual Studio Build Tools 2019, and updated setup tools using **pip install --upgrade setuptools**

I ran the **pip install mod_wsgi** both on Windows CMD and Developer Command Prompt for VS2019

Full Error Message:

ERROR: Command errored out with exit status 1:
     command: 'c:\users\darrius\appdata\local\programs\python\python38\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Darrius\\AppData\\Local\\Temp\\pip-install-qivvz49q\\mod-wsgi\\setup.py'"'"'; __file__='"'"'C:\\Users\\Darrius\\AppData\\Local\\Temp\\pip-install-qivvz49q\\mod-wsgi\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Darrius\AppData\Local\Temp\pip-record-kgy8oddz\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\darrius\appdata\local\programs\python\python38\Include\mod-wsgi'
         cwd: C:\Users\Darrius\AppData\Local\Temp\pip-install-qivvz49q\mod-wsgi\
    Complete output (26 lines):
    c:\users\darrius\appdata\local\programs\python\python38\lib\distutils\dist.py:274: UserWarning: Unknown distribution option: 'bugtrack_url'
      warnings.warn(msg)
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.8
    creating build\lib.win-amd64-3.8\mod_wsgi
    copying src\__init__.py -> build\lib.win-amd64-3.8\mod_wsgi
    creating build\lib.win-amd64-3.8\mod_wsgi\server
    copying src\server\apxs_config.py -> build\lib.win-amd64-3.8\mod_wsgi\server
    copying src\server\environ.py -> build\lib.win-amd64-3.8\mod_wsgi\server
    copying src\server\__init__.py -> build\lib.win-amd64-3.8\mod_wsgi\server
    creating build\lib.win-amd64-3.8\mod_wsgi\server\management
    copying src\server\management\__init__.py -> build\lib.win-amd64-3.8\mod_wsgi\server\management
    creating build\lib.win-amd64-3.8\mod_wsgi\server\management\commands
    copying src\server\management\commands\runmodwsgi.py -> build\lib.win-amd64-3.8\mod_wsgi\server\management\commands
    copying src\server\management\commands\__init__.py -> build\lib.win-amd64-3.8\mod_wsgi\server\management\commands
    creating build\lib.win-amd64-3.8\mod_wsgi\docs
    copying docs\_build\html\__init__.py -> build\lib.win-amd64-3.8\mod_wsgi\docs
    creating build\lib.win-amd64-3.8\mod_wsgi\images
    copying images\__init__.py -> build\lib.win-amd64-3.8\mod_wsgi\images
    copying images\snake-whiskey.jpg -> build\lib.win-amd64-3.8\mod_wsgi\images
    running build_ext
    building 'mod_wsgi.server.mod_wsgi' extension
    error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\darrius\appdata\local\programs\python\python38\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Darrius\\AppData\\Local\\Temp\\pip-install-qivvz49q\\mod-wsgi\\setup.py'"'"'; __file__='"'"'C:\\Users\\Darrius\\AppData\\Local\\Temp\\pip-install-qivvz49q\\mod-wsgi\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Darrius\AppData\Local\Temp\pip-record-kgy8oddz\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\darrius\appdata\local\programs\python\python38\Include\mod-wsgi' Check the logs for full command output.

**Comment by Darrius-Me** [comment]
i got the same problem as #317 but i am not quite satisfied on what its considered solution is since it's not a proper installation and might cause a problem in the future

**Comment by Darrius-Me** [comment]
I also somewhat have the same issue as https://github.com/benfred/implicit/issues/76 but seems like just downloading VC++ 14.0 solved the issue.

**Comment by GrahamDumpleton** [comment]
The error clearly says that you are missing the VC++ 14.0 compiler, or if you have it installed, it is not installed correctly such that is picked up in the environment.

```
error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/
```

So do you have it installed or not?

**Comment by Darrius-Me** [comment]
Well i do have 
Mircrosoft Visual C++ 2015-2019 Redistributable (x86 and x64) 14.22.27281.0
Visual Studio Build Tools 2019

**Comment by GrahamDumpleton** [comment]
Use the command shell created from the compiler environment menu entries to ensure that the environment variables are set correctly so that Python knows where to find the compiler.

If you are using some custom shell, beyond that suggestion, can only suggest you ask on StackOverflow as I don't know how to install the compiler and set up your environment properly so the compiler is found.

When asking on StackOverflow, say you are trying to pip install a Python package with C extension and pip can't find the compiler. This is not mod_wsgi specific and if you highlight mod_wsgi you likely will not get an answer. This issue will happen with any Python package you install which has a C extension component that requires the compiler.

**Comment by Darrius-Me** [comment]
Forgive my assumption on referring the issue on mod_wsgi.

I tried manually installing mod_wsgi (renaming **mod_wsgi.cp36-win_amd64.pyd** to wod_wsgi.so) but there was an error loading the module.

I started a thread on StackOverflow and someone pointed out that it does not "see" my VStudio installation.

I loaded a restore point before installing python and I reinstalled all of the components, VStudio, MS VS C++, python, wheel, etc. and it finally worked.
